### PR TITLE
posix: sched: Implement sched_setparam() and sched_setscheduler()

### LINKED ISSUE #66962: posix: implement `sched_setparam()`
This is part of the See [POSIX Roadmap for LTSv3](https://github.com/zephyrproject-rtos/zephyr/issues/51211) (RFC #51211).

`sched_setparam()` is required as part of `_POSIX_PRIORITY_SCHEDULING` Option Group.

For more information, please refer to https://pubs.opengroup.org/onlinepubs/9699919799/functions/sched_setparam.html

### LINKED ISSUE #66963: posix: implement `sched_setscheduler()`
This is part of the See [POSIX Roadmap for LTSv3](https://github.com/zephyrproject-rtos/zephyr/issues/51211) (RFC #51211).

`sched_setscheduler()` is required as part of `_POSIX_PRIORITY_SCHEDULING` Option Group.

For more information, please refer to https://pubs.opengroup.org/onlinepubs/9699919799/functions/sched_setscheduler.html

### PR DESCRIPTION
**Pull request by moonlight83340**
Implement sched_setparam() and sched_setscheduler() POSIX APIs
as a part of PSE53 _POSIX_PRIORITY_SCHEDULING option group.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/66963
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/66962

### CODE REVIEW

File: tests/posix/common/src/pthread.c [hunk 1]
```diff
@@ -405,6 +405,28 @@ ZTEST(pthread, test_sched_getscheduler)
 
 	zassert_true((rc == -1 && err == ENOSYS));
 }
+ZTEST(pthread, test_sched_setparam)
+{
+	struct sched_param param;
+
+	param.sched_priority = 2;
```
Comment (ycsin) [inline tests/posix/common/src/pthread.c:412]: can probably use static initialization

```suggestion
	struct sched_param param = {
		.sched_priority = 2,
	};
```

**Review by ycsin — COMMENTED** [review]
These functions should also be marked as supported in the [doc](https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/services/portability/posix/option_groups/index.rst) in another commit

I wonder if it is better to move the `sched_*` API tests out of `pthread.c`? cc @cfriedt

**Comment by moonlight83340** [comment]
> These functions should also be marked as supported in the [doc](https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/services/portability/posix/option_groups/index.rst) in another commit
> 
> I wonder if it is better to move the `sched_*` API tests out of `pthread.c`? cc @cfriedt

I wasn't sure on the doc part because getparam() and getscheduler() haven't been update.

```
    sched_get_priority_max(),yes
    sched_get_priority_min(),yes
    sched_getparam(),
    sched_getscheduler(),
    sched_rr_get_interval(),
    sched_setparam(),yes
    sched_setscheduler(),yes
    sched_yield(),yes
```

**Comment by ycsin** [comment]
> I wasn't sure on the doc part because getparam() and getscheduler() haven't been update.

Yeah you will need to update it manually as you implement those missing functions

**Comment by moonlight83340** [comment]
> > I wasn't sure on the doc part because getparam() and getscheduler() haven't been update.
> 
> Yeah you will need to update it manually as you implement those missing functions

done !

**Review by cfriedt — APPROVED** [review]
LGTM

**Comment by moonlight83340** [comment]
> LGTM

Thanks,
Should we add the doc for the getparam and get sheduler part on another pr ?

**Comment by ycsin** [comment]
> > LGTM
> 
> Thanks, Should we add the doc for the getparam and get sheduler part on another pr ?

cc @golowanow

**Review by ycsin — APPROVED** [review]

**Pull request merged**